util/buildRelease/gen_release

```diff
@@ -88,21 +88,18 @@ if (exists($ENV{"CHPL_GEN_RELEASE_NO_CLONE"})) {
     print "[gen_release] Confirm final Git source version used.\n";
     SystemOrDie("cd $rootdir && git --no-pager status && git --no-pager log -1");
 
-    # Leaving this here for now, but this has been replaced by CMake logic
-    # to write the BUILD_VERSION value into config.h
-    # if ($version eq "" or $version eq "developer") {
-    #   print "[gen_release] Create BUILD_VERSION file.\n";
-    #   SystemOrDie("cd $rootdir/compiler && make main/BUILD_VERSION");
-    # }
 
+
+    # write the git-sha while we have the .git directory
+    SystemOrDie("cd $rootdir && util/config/write-git-sha frontend/lib/util --build-version --chpl-home=$rootdir");
+    # use git archive to copy all the files under source control into the archive directory.
     print "[gen_release] Creating build workspace with git archive...\n";
     SystemOrDie("cd $rootdir && git archive --format=tar HEAD | (cd $archive_dir && tar -xf -)");
-
-    if ($version eq "" or $version eq "developer") {
-      print "[gen_release] Copying BUILD_VERSION file.\n";
-      SystemOrDie("util/config/write-git-sha frontend/lib/util --build-version --chpl-home=$archive_dir");
-      SystemOrDie("cp $rootdir/compiler/main/BUILD_VERSION $archive_dir/compiler/main/BUILD_VERSION");
-    }
+    # explicitly copy the BUILD_VERSION file and git-version.cpp files
```

Comment:
I thought that we had replaced the `BUILD_VERSION` file with a Cmake variable (`CHPL_BUILD_VERSION`) and a `#define` in `config.h` (`CHPL_BUILD_VERSION`).

So, I am curious, are we actually using the `BUILD_VERSION` file at this point? I see that it is generated by `write-git-sha` both here and in `CMakeLists.txt`. However, it looks to me like we are only using it in `versionhelp.sh` and `versionhelp-chpldoc.sh` which support testing. (And, we could potentially be using it in other testing scripts that aren't in this repository -- I did not check for those).

Nonetheless, I'm comfortable leaving follow-on work relating to the `BUILD_VERSION` file to another PR as I understand that this PR is just changing when it is copied. Mainly I just wanted to check if you were aware of it.